### PR TITLE
Improve hset and hdel performance with RedisPipeline.

### PR DESCRIPTION
**Pull request by liuh-80**
#### Why I did it
1.Change redis hmset command to hset because HMSET deprecated after redis 4.0.0:
https://redis.io/commands/hmset/

2.Improve DBConnector::hmset and DBConnector::del method performance.
Currently swsscommon implement hmset and del by LUA script, however change to use RedisPipeline can improve performance by 20%:

RedisPipeline hmset: 0.228743
LUA hmset: 0.260371

#### How I did it
Change hmset and del to use RedisPipeline.

#### How to verify it
Pass all test case.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

**Comment by liuh-80** [comment]
This PR just check-in test code, according to discussion result, if we decide to have this improve, we also need improve hdel and remove the lua script.

File: common/rediscommand.h [hunk 1]
```diff
@@ -31,11 +31,11 @@ class RedisCommand {
 #ifndef SWIG
     __attribute__((deprecated))
 #endif
-    void formatHMSET(const std::string &key,
+    void formatHSET(const std::string &key,
                      const std::vector<FieldValueTuple> &values);
 
     template<typename InputIterator>
-    void formatHMSET(const std::string &key,
```
Comment (liuh-80) [inline common/rediscommand.h:38]: formatHMSET is internal API, only use by sonic-swss-common project;
https://github.com/search?q=org%3AAzure+formatHMSET&type=code

**Comment by liuh-80** [comment]
Build sairedis randomly failed, but seems not related with what change in this PR, because other PR also failed:
2022-07-13T08:42:43.9682242Z [  FAILED  ] 2 tests, listed below:
2022-07-13T08:42:43.9682552Z [  FAILED  ] LegacyFdbEntry.fdb_entry_get
2022-07-13T08:42:43.9682874Z [  FAILED  ] Legacy.switch_get

File: common/rediscommand.h [hunk 2]
```diff
@@ -66,12 +69,12 @@ class RedisCommand {
...
 
 template<typename InputIterator>
-void RedisCommand::formatHMSET(const std::string &key,
+void RedisCommand::formatHSET(const std::string &key,
                  InputIterator start, InputIterator stop)
 {
     if (start == stop) throw std::invalid_argument("empty values");
 
-    const char* cmd = "HMSET";
+    const char* cmd = "HSET";
```
[the remark this answers is not in the document]
Comment (qiluo-msft) [reply common/rediscommand.h:77]: Thanks for the comment! Could you include it in either title or PR description?

**Review by qiluo-msft — APPROVED** [review]

File: common/rediscommand.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (liuh-80) [reply common/rediscommand.h:77]: Fixed. description updated.

File: common/dbconnector.cpp [hunk 3]
```diff
@@ -10,7 +10,8 @@
 
 #include "common/dbconnector.h"
 #include "common/redisreply.h"
-#include "common/redisapi.h"
+//#include "common/redisapi.h"
```
Comment (qiluo-msft) [inline common/dbconnector.cpp:13]: Remove dead code.

Comment (liuh-80) [reply common/dbconnector.cpp]: Fixed.

File: common/redispipeline.h [hunk 4]
```diff
@@ -61,6 +62,26 @@ class RedisPipeline {
         return r.release();
     }
 
+    redisReply *pushHSET(const std::string& key, const std::map<std::string, std::string> &values)
```
Comment (qiluo-msft) [inline common/redispipeline.h:65]: >pushHSET [](http://example.com/codeflow?start=16&length=8)

Suggest move the functionality in RedisCommand class.

Comment (liuh-80) [reply common/redispipeline.h]: Fixed, moved to RedisCommand class.

File: common/redispipeline.h [hunk 5]
```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <string>
+#include <queue>
```
Comment (liuh-80) [inline common/redispipeline.h:4]: RedisPipeline using std::queue but not include the header file.
So some time when include this header file will report build error.

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**